syntax = "proto3";

// 完全还原的嵌套消息 SensorInfo 定义
// 其中的字段名已根据 Protobuf 风格指南转换为 snake_case
message SensorInfo {
  string name = 1;
  string vendor = 2;
  int32 version = 3;
  int32 type = 4;
  float max_range = 5;
  float resolution = 6;
  float power = 7;
  int32 min_delay = 8;
}

// 包含完整 SensorInfo 定义的最终版 DeviceInfo 消息
message DeviceInfo {
  string sdkver = 1;
  string app_id = 2;
  string app_version = 3;
  string app_version_code = 4;
  string mid = 5;
  string chid = 6;
  int64 fts = 7;
  string buvid_local = 8;
  int32 first = 9;
  string proc = 10;
  string net = 11;
  string band = 12;
  string osver = 13;
  int64 t = 14;
  int32 cpucount = 15;
  string model = 16;
  string brand = 17;
  string screen = 18;
  string cpumodel = 19;
  string btmac = 20;
  int64 boot = 21;
  string emu = 22;
  string oid = 23;
  string network = 24;
  int64 mem = 25;
  string sensor = 26;
  int64 cpufreq = 27;
  string cpuvendor = 28;
  string sim = 29;
  int32 brightness = 30;
  map<string, string> props = 31;
  map<string, string> sys = 32;
  string wifimac = 33;
  string adid = 34;
  string os = 35;
  string imei = 36;
  string cell = 37;
  string imsi = 38;
  string iccid = 39;
  int32 camcnt = 40;
  string campx = 41;
  int64 totalspace = 42;
  string axposed = 43;
  string maps = 44;
  string files = 45;
  string virtual = 46;
  string virtualproc = 47;
  string gadid = 48;
  string glimit = 49;
  string apps = 50;
  string guid = 51;
  string uid = 52;
  int32 root = 53;
  string camzoom = 54;
  string camlight = 55;
  string oaid = 56;
  string udid = 57;
  string vaid = 58;
  string aaid = 59;
  string androidapp20 = 60;
  int32 androidappcnt = 61;
  string androidsysapp20 = 62;
  int32 battery = 63;
  string batterystate = 64;
  string bssid = 65;
  string build_id = 67;
  string countryiso = 68;
  int64 free_memory = 70;
  string fstorage = 71;
  string kernel_version = 74;
  string languages = 75;
  string mac = 76;
  string ssid = 79;
  int32 systemvolume = 80;
  string wifimaclist = 81;
  int64 memory = 82;
  string str_battery = 83;
  bool is_root = 84;
  string str_brightness = 85;
  string str_app_id = 86;
  string ip = 87;
  string user_agent = 88;
  string light_intensity = 89;
  repeated float device_angle = 90;
  int64 gps_sensor = 91;
  int64 speed_sensor = 92;
  int64 linear_speed_sensor = 93;
  int64 gyroscope_sensor = 94;
  int64 biometric = 95;
  repeated string biometrics = 96;
  int64 last_dump_ts = 97;
  string location = 98;
  string country = 99;
  string city = 100;
  int32 data_activity_state = 101;
  int32 data_connect_state = 102;
  int32 data_network_type = 103;
  int32 voice_network_type = 104;
  int32 voice_service_state = 105;
  int32 usb_connected = 106;
  int32 adb_enabled = 107;
  string ui_version = 108;
  repeated string accessibility_service = 109;
  repeated SensorInfo sensors_info = 110; // 现在指向一个已完整定义的 message
  string drmid = 111;
  bool battery_present = 112;
  string battery_technology = 113;
  int32 battery_temperature = 114;
  int32 battery_voltage = 115;
  int32 battery_plugged = 116;
  int32 battery_health = 117;
  repeated string cpu_abi_list = 118;
  string cpu_abi_libc = 119;
  string cpu_abi_libc64 = 120;
  string cpu_processor = 121;
  string cpu_model_name = 122;
  string cpu_hardware = 123;
  string cpu_features = 124;
  string simulator_attr = 125;
}